File: registration/src/main/java/com/heritage/registration/student/model/StudentRegistrationModel.java
package com.heritage.registration.student.model;

import com.heritage.registration.student.valueobject.BioData;
import com.heritage.registration.student.valueobject.IntendingCourse;
import com.heritage.registration.student.valueobject.JambSubject;
import com.heritage.registration.student.valueobject.School;
import com.heritage.registration.valueobject.RequestStatus;
import com.heritage.registration.workflow.RegistrationWorkflow;
import com.heritage.registration.workflow.WorkflowAction;
import lombok.Builder;
import lombok.Data;

import java.io.Serializable;
import java.util.List;

import static com.heritage.registration.workflow.MakerCheckerWorkflow.getAvailableActions;

@Data
@Builder
public class StudentRegistrationModel implements Serializable {
	private Long id;
	private String uuid;

	private RequestStatus status;
	private boolean enabled;

	private String username;

	private String password;

	private String email;

	private String phone;

	private BioData bioData;

	private List<School> schoolsAttended;

	private List<JambSubject> jambSubjects;

	private List<IntendingCourse> intendingCourses;

	private List<WorkflowAction> availableActions;

	public List<WorkflowAction> getAvailableActions() {
		return RegistrationWorkflow.getAvailableActions(status);
	}
}

File: webapp/app/shared/model/student-registration.model.ts
import {BioData, Course, School, StudentRegistrationRequest, Subject} from "../../openapi";

export class StudentRegistrationModel implements StudentRegistrationRequest {
    bioData: BioData;
    email: string;
    intendingCourses: Array<Course>;
    jambSubjects: Array<Subject>;
    password: string;
    phone: string;
    schoolsAttended: Array<School>;
    username: string;

    uuid: string;
    code!: string;
    name!: string;
    shortName!: string;
    enable!: boolean;
    status?: string;
    availableActions!: Array<WorkflowActionResponse>;

    constructor() {
        this.uuid = ''
    }
}

